spark/src/main/scala/org/apache/comet/rules/CometExecRule.scala

```diff
@@ -239,6 +325,11 @@ case class CometExecRule(session: SparkSession) extends Rule[SparkPlan] {
         convertToComet(s, CometShuffleExchangeExec).getOrElse(s)
 
       case op =>
+        // Check if this operator has already been tagged with fallback reasons
+        if (hasExplainInfo(op)) {
+          return op
+        }
```

Comment:
Note that this is also a performance optimization in the case where the rule is applied to a plan multiple times (which does happen with AQE)